packages/commerce-sdk-react/src/hooks/createUseMutation.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {UseMutationResult, DefaultError} from '@tanstack/react-query'
+import {ApiClients, ApiMethod, Argument, DataType, CacheUpdateGetter} from './types'
+import useCommerceApi from './useCommerceApi'
+import {useMutation} from './useMutation'
+
+/**
+ * Type to identify client methods that are functions
+ */
+export type MethodsOf<C> = {
+    [K in keyof C]: C[K] extends ApiMethod<any, any> ? K : never
+}[keyof C]
+
+/**
+ * Helper to ensure a method is an ApiMethod
+ */
+type EnsureApiMethod<T> = T extends ApiMethod<any, any> ? T : never
+
+/**
+ * Options for creating a typed mutation hook for a specific API client
+ */
+export interface CreateUseMutationOptions<
+    ClientKey extends keyof ApiClients,
+    Mutation extends MethodsOf<ApiClients[ClientKey]>
+> {
+    /** The client key in ApiClients (e.g. 'shopperSearch', 'shopperProducts') */
+    clientKey: ClientKey
+    /** Function to get the cache updates matrix for the mutation */
+    getCacheUpdates: (mutation: Mutation) => CacheUpdateGetter<any, any> | undefined
```

Comment:
Are the CacheUpdateGetter args types known so we don't need to use any?

Reply:
This requires using type assertion, but I'm not sure if it is possible in typescript `interface`... This is some advanced typescript techniques that I'm not familiar, if you know how to do that, let me know